nrf_modem/doc/CHANGELOG.rst

```diff
@@ -9,6 +9,55 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+nrf_modem 2.5.0
+***************
+
+:ref:`Core library <architecture>`
+==================================
+
+* The DFU result is now returned by the :c:param:`dfu_callback` callback in :c:struct:`nrf_modem_init_params` struct.
+* It is no longer necessary to call :c:func:`nrf_modem_init` twice when applying modem firmware updates.
+* Added the SoftSIM API to use a software SIM with the nRF modem.
+
+:ref:`Sockets <nrf_sockets>`
+============================
+
+* Added
+
+  * The :c:macro:`NRF_SO_EXCEPTIONAL_DATA` socket option to enable sending data as part of exceptional events (3GPP).
+  * The :c:macro:`MSG_WAITACK` flag to request a blocking send operation until the request is acknowledged by the network.
+  * Enhanced APN rate control.
+
+* Removed:
+
+  * The :c:param:`sa_len`, :c:param:`sin_len` and :c:param:`sin6_len` from the :c:struct:`nrf_sockaddr` :c:struct:`nrf_sockaddr_in` and :c:struct:`nrf_sockaddr_in6` structs, respectively.
+
+* Replaced the :c:macro:`NRF_SO_BINDTODEVICE` socket option with :c:macro:`NRF_SO_BINDTOPDN`. The new option takes an integer for the PDN ID.
+
+
+:ref:`AT interface <nrf_modem_at>`
+==================================
+
+* Added the option to set a timeout for the time to wait for ongoing AT commands to complete by calling the :c:func:`nrf_modem_at_sem_timeout_set` function.
+
+:ref:`GNSS interface <nrf_modem_gnss>`
+======================================
+
+* Added:
+
+  * Support for QZSS assistance.
```

Comment:
do we need documentation for QZSS assistance?

Reply:
@tokangas

Reply:
The GNSS interface conceptual documentation needs to be updated due to A-GPS -> A-GNSS changes. I have one other improvement I was planning to do there, so I can handle the update at the same time. @divipillai Is it ok to handle the doc update separately?

Reply:
I am okay with considering this separately, but do we plan to have the doc update in this release (2.5 release)?

Reply:
Yes, we do. I'm working on it.

Reply:
Thanks for the update!

Reply:
@divipillai I created a couple of PRs to update GNSS related documentation:
https://github.com/nrfconnect/sdk-nrfxlib/pull/1080
https://github.com/nrfconnect/sdk-nrf/pull/12459